pulsar-io/cassandra/src/test/java/org/apache/pulsar/io/cassandra/CassandraSinkExec.java

```diff
@@ -0,0 +1,98 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.pulsar.io.cassandra;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import org.apache.pulsar.common.io.SinkConfig;
+import org.apache.pulsar.functions.LocalRunner;
+import org.apache.pulsar.io.cassandra.producers.InputTopicProducerThread;
+import org.apache.pulsar.io.cassandra.producers.InputTopicStringProducer;
+import org.apache.pulsar.io.cassandra.producers.ReadingSchemaRecordProducer;
+import org.yaml.snakeyaml.Yaml;
+
+/**
+ * Useful for testing within IDE.
+ *
+ */
+public class CassandraSinkExec {
+
+    public static final String BROKER_URL = "pulsar://localhost:6650";
+    public static final String INPUT_TOPIC = "persistent://public/default/air-quality-reading-generic";
+    // public static final String INPUT_TOPIC = "persistent://public/default/air-quality-reading-string";
+    public static final String CONFIG_FILE = "cassandra-sink-config.yaml";
+
+    public static void main(String[] args) throws Exception {
+
+        SinkConfig config = getSinkConfig();
+
+        final LocalRunner localRunner =
+                LocalRunner.builder()
+                        .brokerServiceUrl(BROKER_URL)
+                        .sinkConfig(config)
+                        .build();
+
+        localRunner.start(false);
+
+        sendData();
+        TimeUnit.MINUTES.sleep(10);
+
+        localRunner.stop();
+
+        System.exit(0);
+    }
+
+    private static SinkConfig getSinkConfig() throws FileNotFoundException {
+        SinkConfig sinkConfig = SinkConfig.builder()
+                .autoAck(true)
+                .cleanupSubscription(Boolean.TRUE)
+                .configs(getConfigs())
+                .className(CassandraGenericRecordSink.class.getName())
+                .inputs(Collections.singletonList(INPUT_TOPIC))
+                .name("CassandraSink")
+                .build();
+
+        return sinkConfig;
+    }
+
+    private static Map<String, Object> getConfigs() throws FileNotFoundException {
+        Map<String, Object> configs = new HashMap<String, Object>();
+
+        ClassLoader classLoader = Thread.currentThread().getContextClassLoader();
+        File file = new File(classLoader.getResource(CONFIG_FILE).getFile());
+
+        Yaml yaml = new Yaml();
+        configs = yaml.load(new FileInputStream(file));
+
+        return configs;
+    }
+
+    private static final void sendData() throws InterruptedException {
+        TimeUnit.SECONDS.sleep(10);
```

Comment:
do we need to sleep a fixed amount of time ?

Reply:
The only reason I sleep 10 seconds, is so I can easily visualize the data as it is published to the system. The delay allows me time to execute a query against Cassandra in order to detect the new records.